Serializer/PlainTextDeserializationVisitor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Smartbox\CoreBundle\Serializer;
+
+use JMS\Serializer\AbstractVisitor;
+use JMS\Serializer\Context;
+use JMS\Serializer\Exception\RuntimeException;
+use JMS\Serializer\GraphNavigator;
+use JMS\Serializer\Metadata\ClassMetadata;
+use JMS\Serializer\Metadata\PropertyMetadata;
+
+/**
+ * This is a really simple visitor that will effectively bypass JMS. Useful for REST APIs that are not truly REST APIs
+ * and return plain text responses, which JMS by default refuses to process because they are not real JSON.
+ */
+class PlainTextDeserializationVisitor extends AbstractVisitor
+{
+    public function visitNull($data, array $type, Context $context)
+    {
+        throw new RuntimeException('PlainTextDeserializationVisitor cannot visit Null types.');
+    }
+
+    public function visitString($data, array $type, Context $context)
+    {
+        return $data;
```

Comment:
I think here we should cast data to string, so that the response will always be a string
```
return (string) $data;
```

Reply:
Done